test/unit/test-amp-story-player.js

```diff
@@ -385,6 +389,90 @@ describes.realWin('AmpStoryPlayer', {amp: false}, (env) => {
       );
     });
 
+    it('adds stories programmatically', async () => {
+      buildStoryPlayer(3);
+      await manager.loadPlayers();
+
+      const storyObjects = createStoryObjects(1);
+      playerEl.add(storyObjects);
+
+      const stories = toArray(playerEl.querySelectorAll('a'));
+
+      expect(stories.length).to.eql(4);
+    });
+
+    it.only('throws an error if the add method is not passed an array', async () => {
+      buildStoryPlayer();
+      await manager.loadPlayers();
+
+      return expect(() => playerEl.add(7)).to.throw(
+        '"stories" parameter has the wrong structure'
+      );
+    });
+
+    it.only('throws an error if the add method is passed an array with incorrectly structured story objects', async () => {
+      buildStoryPlayer();
+      await manager.loadPlayers();
+
+      const wrongStoryObjects = [{notHref: true}];
+
+      return expect(() => playerEl.add(wrongStoryObjects)).to.throw(
+        '"stories" parameter has the wrong structure'
+      );
+    });
+
+    it('adds no stories when sending an empty array of new stories', async () => {
+      buildStoryPlayer(3);
+      await manager.loadPlayers();
+
+      const storyObjects = createStoryObjects(0);
+      playerEl.add(storyObjects);
+
+      const stories = toArray(playerEl.querySelectorAll('a'));
+
+      expect(stories.length).to.eql(3);
+    });
+
+    it(
+      'creates and assigns iframes to added stories when there are ' +
+        'less than the maximum iframes set up',
+      async () => {
+        buildStoryPlayer();
+        await manager.loadPlayers();
+
+        const storyObjects = createStoryObjects(2);
+        playerEl.add(storyObjects);
+
+        const stories = toArray(playerEl.querySelectorAll('a'));
+
+        expect(stories[0][IFRAME_IDX]).to.not.be.undefined;
+        expect(stories[1][IFRAME_IDX]).to.not.be.undefined;
+        expect(stories[2][IFRAME_IDX]).to.not.be.undefined;
+      }
+    );
+
+    it(
+      'assigns an existing iframe to the first added story when the current ' +
+        'story is the last one, and the maximum number of iframes has been set up',
+      async () => {
+        buildStoryPlayer(3);
+        await manager.loadPlayers();
+
+        swipeLeft();
+        swipeLeft();
+
+        win.requestAnimationFrame(() => {
```

Comment:
The test will return after this line and before running all the code within the callback. I think Enrique wrote a new `afterRenderPromise` helper that you could use here.

Reply:
Done in latest commits